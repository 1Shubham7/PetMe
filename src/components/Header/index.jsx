import React from 'react';
import { Link } from 'react-router-dom';
import buttons from '../../helpers/buttonsForHeader';
import logo from '../../assets/images/logo.jpg';

function Header() {
  return (
    <header
      id="header"
      className="max-w-screen text-white bg-orange-900 flex items-center justify-around lg:justify-around"
    >
      <div className="flex items-center md:items-center p-5" id="logo">
        <span className="p-2"> Saving Lives </span>
        <a
          href="/"
        >
          <img className="rounded-full w-20" src={logo} alt="logo" />
        </a>
        <span className="p-2"> Saving Animals </span>
      </div>

      <nav className="flex items-center justify-center">
        { buttons.map((button) => (
          <a
            key={button.id}
            href={button.link}
            className="p-2 hover:text-orange-300 hover:underline"
          >
            {button.name}
          </a>
        ))}

        <Link
          to="/contact"
          className="p-2 hover:text-orange-300 hover:underline"
        >
          <button
            type="button"
            className="bg-[#C57837] p-4 focus:outline-none font-bold rounded mt-4 md:mt-0"
          >
            Contact Us
          </button>

        </Link>
      </nav>

    </header>
  );
}

export default Header;
