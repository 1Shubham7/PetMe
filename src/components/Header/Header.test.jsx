import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '.';

const NUMBER_OF_LINKS = 10;

describe('Header', () => {
  test('renders Header', () => {
    const { getByText } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const linkElement = getByText(/saving lives/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('check if buttons and links of header are rendered', () => {
    const { getAllByRole } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );

    const linkElement = getAllByRole('button');
    expect(linkElement).toHaveLength(1);

    const linkElement2 = getAllByRole('link');
    expect(linkElement2).toHaveLength(NUMBER_OF_LINKS);
  });
});
