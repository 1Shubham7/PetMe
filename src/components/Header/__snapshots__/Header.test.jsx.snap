// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders Header with snapshot 1`] = `
<DocumentFragment>
  <header
    class="max-w-screen text-white bg-orange-900 flex items-center justify-around lg:justify-around"
    id="header"
  >
    <div
      class="flex items-center md:items-center p-5"
      id="logo"
    >
      <span
        class="p-2"
      >
         Saving Lives 
      </span>
      <a
        href="/"
      >
        <img
          alt="logo"
          class="rounded-full w-20"
          src="/src/assets/images/logo.jpg"
        />
      </a>
      <span
        class="p-2"
      >
         Saving Animals 
      </span>
    </div>
    <nav
      class="flex items-center justify-center"
    >
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/"
      >
        Home
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/about"
      >
        About
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/contact"
      >
        Contact
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/petnews"
      >
        PetNews
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/blogs"
      >
        Blogs
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/donateanimals"
      >
        Donate Animals
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/donatemoney"
      >
        Suport Us
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/sosreport"
      >
        SOS Report
      </a>
      <a
        class="p-2 hover:text-orange-300 hover:underline"
        href="/contact"
      >
        <button
          class="bg-[#C57837] p-4 focus:outline-none font-bold rounded mt-4 md:mt-0"
          type="button"
        >
          Contact Us
        </button>
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
